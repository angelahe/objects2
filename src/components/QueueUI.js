import React from "react";
import LIFOqueue from "./LIFOqueue";
import FIFOqueue from "./FIFOqueue";
import QueueItem from "./QueueItem";

import '../styles/Styles140.css';
import addbtn from '../images/add_FFFFFF.png';
import forward from "../images/forward2_FFFFFF.png";

class QueueUI extends React.Component {

  constructor() {
    super()
    this.state = {
      queue1: new LIFOqueue(),
      queue2: new FIFOqueue(),
      queue1out: "",
      queue2out: ""
    }
  }

  onBtnAdd1Click = () => {
    const name = document.getElementById("queue1name").value;
    console.log("add to this queue ", name);
    this.state.queue1.addToQueue(name);
    this.setState({queue1: this.state.queue1});
    document.getElementById("queue1name").value = "";
  }

  onBtnAdd2Click = () => {
    const name = document.getElementById("queue2name").value;
    console.log("add to 2nd queue ", name);
    this.state.queue2.addToQueue(name);
    this.setState({queue2: this.state.queue2});
    document.getElementById("queue2name").value = "";
  }

  onNextQueue1Click = () => {
    const nextFromQueue = this.state.queue1.popFromQueue();
    this.setState({queue1out: nextFromQueue });

  }

  onNextQueue2Click = () => {
    const nextFromQueue = this.state.queue2.firstFromQueue();
    this.setState({queue2out: nextFromQueue});
  }

  render() {

    const queue1items = this.state.queue1.queue.map((person) =>
      <QueueItem key={person} person={person}/>
    );
    const queue2items = this.state.queue2.queue.map((person) =>
      <QueueItem key={person} person={person}/>
    );

    return(
      <div>
        <h1>Hello from LIFO and FIFO</h1>
        <div className = "AppContainer">
          <div className = "AppPanel">
            <h2>LIFO Queue</h2>
            <div className = "ItemBox AppHeader">
              <span className = "AddItem">Add to Queue</span>
              <input className="InputText" id="queue1name" placeholder="next into queue"></input><br/>
              <button className="AppBtn" onClick={this.onBtnAdd1Click}>
                <img className="btnImg" src={addbtn} alt="Add"/>
              </button>
            </div>
            <div className = "AppList">
              {queue1items}
            </div>
            <div className = "ItemBox AppHeader">
              <span className = "AddItem">Pull from Queue:</span>
              <span className = "AddItem">{this.state.queue1out}</span><br/>
              <button className="AppBtn" onClick={this.onNextQueue1Click}>
                <img className="btnImg" src={forward} alt="Add"/>
              </button>
            </div>
        </div>
          <div className = "AppPanel">
            <h2>FIFO queue</h2>
            <div className = "ItemBox AppHeader">
              <span className = "AddItem">Add to Queue</span>
              <input className="InputText" id="queue2name" placeholder="next into queue"></input><br/>
              <button className="AppBtn" onClick={this.onBtnAdd2Click}>
                <img className="btnImg" src={addbtn} alt="Add"/>
              </button>
            </div>
            <div className = "AppList">
              {queue2items}
            </div>
            <div className = "ItemBox AppHeader">
              <span className = "AddItem">Pull from Queue:</span>
              <span className = "AddItem">{this.state.queue2out}</span><br/>
              <button className="AppBtn" onClick={this.onNextQueue2Click}>
                <img className="btnImg" src={forward} alt="Add"/>
              </button>
            </div>
          </div>
        </div>
      </div>
    )
  }
}

export default QueueUI