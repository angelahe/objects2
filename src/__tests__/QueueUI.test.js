import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import QueueUI from "../components/QueueUI";
import renderer from "react-test-renderer";

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<QueueUI />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders without crashing', () => {
  shallow(<QueueUI />);
});

test('LinkedList renders correctly', () => {
  const tree = renderer.create(<QueueUI />).toJSON();
  expect(tree).toMatchSnapshot();
});