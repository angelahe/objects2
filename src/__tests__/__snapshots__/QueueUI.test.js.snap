// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkedList renders correctly 1`] = `
<div>
  <h1>
    Hello from LIFO and FIFO
  </h1>
  <div
    className="AppContainer"
  >
    <div
      className="AppPanel"
    >
      <div
        className="ItemBox AppHeader"
      >
        <span
          className="AddItem"
        >
          Add to Queue
        </span>
        <input
          className="InputText"
          id="queue1name"
          placeholder="next into queue"
        />
        <br />
        <button
          className="AppBtn"
          onClick={[Function]}
        >
          <img
            alt="Add"
            className="btnImg"
            src="add_FFFFFF.png"
          />
        </button>
      </div>
      <div
        className="AppList"
      />
    </div>
    <div
      className="AppPanel"
    >
      <div
        className="ItemBox AppHeader"
      >
        <span
          className="AddItem"
        >
          Add to Queue
        </span>
        <input
          className="InputText"
          id="queue2name"
          placeholder="next into queue"
        />
        <br />
        <button
          className="AppBtn"
          onClick={[Function]}
        >
          <img
            alt="Add"
            className="btnImg"
            src="add_FFFFFF.png"
          />
        </button>
      </div>
      <div
        className="AppList"
      />
    </div>
  </div>
</div>
`;
